src/main/python/vantiqsdk.py

```diff
@@ -0,0 +1,1399 @@
+        self._subscriber: Union[_VantiqSubscriber, None] = None
+
+    async def __aenter__(self):
+        await self.connect()
+        return self
+
+    async def __aexit__(self, exc_type, exc, tb):
+        await self.close()
+
+    async def connect(self) -> None:
+        """Make the connection to the Vantiq server
+
+        Returns:
+            None
+        Raises:
+            VantiqException for connection issues
+
+        Notes:
+            Should precede the .authenticate().  Must be done before other operations.
+        """
+
+        if not self._is_connected:
+            await self._connection.connect()
+            self._is_connected = True
+            self._vlog.debug('Connected to server: %s', self._server)
+
+    def get_server(self):
+        """Returns the server url used for this connection."""
+        return self._server
+
+    def get_api_version(self):
+        """Returns the API version used for this connection."""
+        return self._api_version
+
+    async def set_access_token(self, access_token: str) -> None:
+        """Set the access token for server access.
+
+        Access to the Vantiq server is controlled.  Access can be granted by using an access token (recommended) or
+        a username and password.
+
+        Parameters:
+            access_token : str
+                Access token to be used to gain access to the Vantiq system
+        """
+        if not self._is_connected:
+            await self.connect()
+        self._access_token = access_token
+        self._auth_header = 'Bearer ' + self._access_token
+        self._is_authenticated = True  # We have an access token, we'll figure we're OK til we're not
+
+    def get_access_token(self) -> str:
+        """Returns the access token currently in use."""
+        return self._access_token
+
+    def set_username(self, username: str) -> None:
+        """Set the username to be used to access the Vantiq server"""
+        self._username = username
+
+    def get_username(self) -> str:
+        """Returns the username currently in use."""
+        return self._username
+
+    def get_id_token(self) -> str:
+        return self._id_token
+
+    def is_authenticated(self) -> bool:
+        """Returns a boolean indicating whether this Vantiq instance is authenticated."""
+        return self._is_authenticated
+
+    async def authenticate(self, username: str, password: str) -> None:
+        """(Async) Use a username and password to connect to the Vantiq system.
+
+        Parameters:
+            username : str
+                The username to be used to authenticate to the Vantiq server.
+            password : str
+                The password to be used to authenticate to the Vantiq server.
+
+        Returns:
+            Raises a VantiqException in case of failure.
+
+        Example:
+        ::
+            await client.authenticate('vantiquser', 'secret')
+        """
+        if not self._is_connected:
```

Comment:
This answers my question above -- `connect` is called implicitly.  Should correct one place that implies it is not.

Reply:
updated